vm/src/builtins/object.rs

```diff
@@ -323,6 +323,19 @@ pub(crate) fn setattr(
 ) -> PyResult<()> {
     vm_trace!("object.__setattr__({:?}, {}, {:?})", obj, attr_name, value);
 
+    if attr_name.as_str() == "__debug__" {
```

Comment:
I'd revert these changes. This adds a check for every single look-up and doesn't match behavior expected (the `setattr` is supposed to succeed in changing the value). 

Looking at the disassembly of looking up `__debug__` it seems as if `True` is immediately loaded, indicating no dynamic look-up is performed for `__debug__`. CPython probably detects `__debug__` being looked-up in their peephole optimizer and replaces it with `True` (that's why I said this change is probably a bit involved).

Reply:
Fair enough. Better left for another PR.